crm_compassion/models/event_compassion.py

```diff
@@ -370,22 +368,22 @@ def onchange_type(self):
     def onchange_start_date(self):
         """ Update end_date and hold_start_date as soon as start_date is
         changed """
-        days_allocate_before_event = self.env['res.config.settings'].sudo().get_param(
+        days_allocate_before_event = self.env[
+            'demand.planning.settings'].get_param(
             'days_allocate_before_event')
         dt = timedelta(days=days_allocate_before_event)
         for event in self.filtered('start_date'):
-            event.hold_start_date = event.start_date - dt
+            event.hold_start_date = (event.start_date - dt).date()
             if not event.end_date or event.end_date < event.start_date:
                 event.end_date = event.start_date
 
     @api.onchange('end_date')
     @api.multi
     def onchange_end_date(self):
-        days_after = self.env['res.config.settings'].sudo().get_param(
+        days_after = self.env['demand.planning.settings'].get_param(
```

Comment:
The class demand.planning.settings was removed, I think this change should be reverted